pkgs/os-specific/linux/kernel/update-mainline.py

```diff
@@ -98,6 +98,15 @@ def commit(message):
     return subprocess.check_call(["git", "commit", "-m", message, VERSIONS_FILE])
 
 
+def to_nixpkgs_version(version: Version) -> str:
```

Comment:
I wonder if it might make more sense to preserve the upstream version, and just use the packaging version where necessary, rather than trying to round-trip the upstream version through packaging and back again?

Reply:
Yeah, I'll do that I think.

Reply:
Rewrote that bit a bunch more